pkg/upgrade/upgrade.go

```diff
@@ -91,24 +91,21 @@ func NodeVersionInfo() (map[string]NodeVersion, error) {
 }
 
 func UpdateInfo(info map[string]NodeVersion, dtk registry.DriverToolkitEntry, imageURL string) (map[string]NodeVersion, error) {
+	dtk.ImageURL = imageURL
```

Comment:
This change will probably fix all the "no FROM image" bugs that we have,no?

Reply:
Not really, this is only moving a duplicated line to a common place. But you just made me realise there might be a bug here and it may lead to an empty FROM without knowing what happened. Let me check and I will open a PR if that is the case.

Reply:
due to moving the duplicated line outside of the "if" based on architecture, you are making sure that ImageURL cannot be empty, and this solves "empty FROM" issue. Do you think that we need all those verifications in that function

Reply:
The image URL for DTK can still be empty, you need a match for the kernel running in the nodes with those in the DTK or else it fails silently. I opened #231 to fix it and get a proper error message.